src/farm/library.cairo

```diff
@@ -547,4 +607,134 @@ namespace CarbonableFarmer {
             contract_address=contract_address, index=next, total_supply=total_supply
         );
     }
+
+    func _create_vestings{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        total_amount: felt
+    ) {
+        alloc_locals;
+
+        let (starkvest_address) = starkvest_address_.read();
+        let (contract_address) = carbonable_project_address_.read();
+        let (total_supply) = IERC721Enumerable.totalSupply(contract_address=contract_address);
+
+        let zero = Uint256(low=0, high=0);
+        let (is_zero) = uint256_eq(total_supply, zero);
+        if (is_zero == TRUE) {
+            return ();
+        }
+
+        let one = Uint256(low=1, high=0);
+        let (index) = SafeUint256.sub_le(total_supply, one);
+        let (token_total_deposited) = total_locked();
+        return _create_vestings_iter(
+            contract_address=contract_address,
+            starkvest_address=starkvest_address,
+            total_amount=total_amount,
+            token_total_deposited=token_total_deposited,
+            index=index,
+            total_supply=total_supply,
+        );
+    }
+
+    func _create_vestings_iter{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        contract_address: felt,
+        starkvest_address: felt,
+        total_amount: felt,
+        token_total_deposited: Uint256,
+        index: Uint256,
+        total_supply: Uint256,
+    ) {
+        alloc_locals;
+
+        // Get registred address of the current token index
+        let (token_id) = IERC721Enumerable.tokenByIndex(
+            contract_address=contract_address, index=index
+        );
+        let (address) = registration_.read(token_id);
+
+        // If registred then create a vesting for each token_id
+        let one = Uint256(low=1, high=0);
+        if (address != 0) {
+            // The value 1 is the 1 token_id
+            let address_token_amount = 1;
+            let (address_amount) = _amount_to_vesting(
+                token_total_deposited=token_total_deposited,
+                total_amount=total_amount,
+                address_token_amount=address_token_amount,
+            );
+
+            // [Interaction] Starkvest - create vesting for address
+            // Vesting, with no cliff period, no duration and no delay to start
+            let beneficiary = address;
+            let cliff_delta = 0;
+            let start = 1;
+            let duration = 1;
+            let slice_period_seconds = 1;
+            let revocable = TRUE;
+
+            // [Interaction] Starkvest - create vesting for address
+            let (vesting_id) = IStarkVest.create_vesting(
+                starkvest_address,
+                beneficiary,
+                cliff_delta,
+                start,
+                duration,
+                slice_period_seconds,
+                revocable,
+                address_amount,
+            );
+
+            // TODO: verify vesting_id is not 0
```

Comment:
Why not 0?
Why not done?

Reply:
Forgot comment to delete, but there's no need to check it here (besides, you don't want to break the recursion function).